src/stack.js

```diff
@@ -1,5 +1,39 @@
 'use strict'
 
 export default class Stack {
-  // your code here
+  constructor(){
+    this.storage = []
+  }
+
+  push(element){
+    return this.storage.push(element)
+  }
+
+  length(){
+    return this.storage.length
+  }
+
+  pop(){
+    if(this.storage.length === 0) {
+      return null
+    } else {
+      return this.storage.pop()
+    }
+  }
+
+  peek() {
+    if(this.storage.length === 0) {
+      return null
+    } else {
+      return this.storage[this.storage.length-1] 
+    }
+  }
+
+  isEmpty() {
+    if(this.storage.length === 0) {
+      return true
+    } else {
+      return false 
+    }
+  }
 }
```

Comment:
Queue
No length function

Linked List 
no .data or .next function

Doubly Linked List 
missing

Hash Table 
missing

Binary Search Tree
missing traverse
missing count

Directed Graph 
missing 

General: It looks good, scan through your program and remove unnecessary whitespace! :)